src/select/Select.tsx

```diff
@@ -44,9 +45,19 @@ function SelectComponent<T extends Option = Option>(
     () => selectRef.current
   );
 
+  useEffect(() => {
+    if (props.options.length !== selectOwnState.options.length) {
```

Comment:
This feels unsafe but I'm not sure if it worths doing a more proper equality check. @Hipo/frontend-team open to any suggestions here.

Reply:
As @Anlerkan suggested, I removed the `options` field from `selectOwnState` and directly set it in the context state from the props with `generateSelectState` function